src/transforms/sort.jl

```diff
@@ -0,0 +1,54 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Sort(by, ascending)
+
+Returns a table copy with rows sorted by values of a specific column.
+The `by` value specifies the column used to sort, it can be a index (Int) or a name (Symbol).
+The `ascending` value is a boolean that specifies the sort order, the default value is true.
+
+
+# Examples
+
+```julia
+Sort(:a)
+Sort(:a, ascending=true)
+Sort(1)
+Sort(1, ascending=false)
+```
+"""
+
+struct Sort <: Stateless
+  by::Union{Int, Symbol}
+  ascending::Bool
+end
+
+Sort(by; ascending=true) = Sort(by, ascending)
+
+isrevertible(::Type{<:Sort}) = true
+
+function apply(transform::Sort, table)
+  # use selected column to calculate new order
+  scol = Tables.getcolumn(table, transform.by)
+  neworder = sortperm(scol, rev=!transform.ascending)
+
+  # sort rows
+  rows = Tables.rowtable(table)
+  rows = rows[neworder]
```

Comment:
Nice I think this is a great approach. We need to collect the rows anyways so using Tables.rowtable seems to make sense. I think Tables.rows is lazy sometimes. If it is not lazy, we should use it instead to use the most efficient representation of rows for the given table type.

Reply:
I dind't find in documentation specifically if its lazy or not. There is a way to check in code?

Reply:
Good question. I don't think we have a mechanism to discover the lazyness programmatically, as far as I remember. If the docstring of Table.rows doesn't require the AbstractArray interface then we probably can't use it.

If Table.rows requires the AbstractArray interface, that means that we can use getindex (i.e. A[inds]) to reorder the array as you did. If it does not require the interface, that means that someone may implement a lazy Table.rows and our code will break, in which case it is safer to always use Tables.rowtable as you implemented already.